common-ts/src/actions/actionHelpers/accountDeletionHelpers.ts

```diff
@@ -0,0 +1,255 @@
+ *
+ * States that affect whether an account can be deleted:
+ * - A user stats account that is older than 13 days => All accounts can be deleted
+ * - A user account that is idle => The user account in question can be deleted
+ * - A user account that can be marked idle => The user account in question can be deleted after sending a markIdle instruction
+ * - A user that is not/can not be idle (yet), with no stats account older than 13 days => wait until the account is idle and then try again. If the user has open perp positions, orders or borrows, they need to be closed first as well.
+ * @param userAccount
+ * @param userStatsAccount
+ */
+const getAccountDeletionStepsToTake = (
+	user: User,
+	userStatsAccount: UserStatsAccount,
+	currentSlot: number
+): AccountDeletionStep[] => {
+	const userAccount = user.getUserAccount();
+	const statsAccountIsPastDeletionCutoff =
+		getStatsAccountIsPastDeletionCutoff(userStatsAccount);
+
+	// Account is past deletion cutoff
+	if (statsAccountIsPastDeletionCutoff) {
+		return ['sendAccountDeletionIx'];
+	}
+
+	// Account needs to be unwound first
+	const hasOpenPositionsOrOrders = accountHasOpenPositionsOrOrders(user);
+
+	if (hasOpenPositionsOrOrders) {
+		return ['askToCloseAllPositionsOrdersBorrows'];
+	}
+
+	// Account is idle and can be deleted
+	const userAccountIsIdle = userAccount.idle;
+
+	if (userAccountIsIdle) {
+		return ['sendAccountDeletionIx'];
+	}
+
+	// Account can be marked idle and then deleted
+	const canBeMarkedIdle = user.canMakeIdle(new BN(currentSlot));
+
+	if (canBeMarkedIdle) {
+		return ['sendTriggerAccountIdleIx', 'sendAccountDeletionIx'];
+	}
+
+	// Account is not idle and can not be marked idle yet, wait until it is (or can be marked) idle and then try again
+	return ['askToWait'];
+};
+
+/**
+ * Builds the necessary transaction to delete a user account. This method will throw an error if the account cannot be deleted instantly.
+ * @param driftClient
+ * @param user
+ * @param userStatsAccount
+ * @param latestSlot
+ * @returns
+ */
+const tryDeleteUserAccount = async (
+	driftClient: DriftClient,
+	user: User,
+	userStatsAccount: UserStatsAccount,
+	latestSlot: number
+) => {
+	const canBeDeleted = getAccountCanBeDeletedInstantly(
+		user,
+		userStatsAccount,
+		latestSlot
+	);
+
+	if (canBeDeleted === 'no') {
+		throw new Error('Account cannot be deleted');
+	}
+
+	const Ixs: TransactionInstruction[] = [];
+
+	if (canBeDeleted === 'yes-after-making-idle') {
+		// Create the make idle instruction
+		Ixs.push(
+			await driftClient.getUpdateUserIdleIx(
+				user.userAccountPublicKey,
+				user.getUserAccount()
+			)
+		);
+	}
+
+	// Create the delete account instruction
+	Ixs.push(await driftClient.getUserDeletionIx(user.userAccountPublicKey));
```

Comment:
you abstracted this ix creation fn out of the SDK, but I think the subsequent success handling is missing and should probably be added to this function too

https://github.com/drift-labs/protocol-v2/pull/758/files#diff-24da82cf4077e0e816f565f4f35b572708ac94147b9ab847ba5c77f1cc508397R1148